### ISSUE TITLE
False negative using --strict-equality and None

### ISSUE THREAD

**Issue by Tinche** [python/mypy#14541]
Howdy.

I'm using strict equality, and while refactoring some code ran into this issue.

Example:
```python
a: str = ""
if a is None:
    pass
```

```
$ mypy --strict-equality --warn-unreachable a05.py
Success: no issues found in 1 source file

$ mypy --version
mypy 0.991 (compiled: yes)
$ python --version
Python 3.11.1
```

I would expect this to be an error; this is never going to be true and we're performing completely unnecessary work, right? I would also expect for the body of the `if` statement to be an unreachable error, I guess.

**Comment by thomkeh** [comment]
I'm getting an unreachable error here: https://mypy-play.net/?mypy=latest&python=3.11&flags=strict%2Cwarn-unreachable&gist=929e882f681a1938b9495686c67b72f2

It seems it just doesn't work with `pass`?

**Comment by Tinche** [comment]
Interesting. This is also a false negative:

```python
a: str = ""
if a is None:
    raise Exception()  # No error, wrong
```
But this works:
```python
a: str = ""
if a is None:
    1 + 1  # Error, correct
```

**Comment by Hnasar** [comment]
I also ran into this false negative between `--strict-equality` and `--unreachable` when comparing with `None`.

`--strict-equality` fails to detect an error when comparing with `None`, and although `warn-unreachable` is able to properly error on `a is None`; it fails if doing `a == None`. 

Here's a playground that reproduces the false negatives with mypy 1.11.2
https://mypy-play.net/?mypy=latest&python=3.12&flags=strict-equality&gist=9d249248acab3dae4a0e3e486d560321

**Comment by zzJinux** [comment]
Unfortunately, this is by design: https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-strict-equality

We should file a feature request. This is too unintuitive feature for static type checkers
